assets/setup-wizard/newsletter/signup-form.js

```diff
@@ -22,43 +26,82 @@ import { useSetupWizardStep } from '../data/use-setup-wizard-step';
 const SignupForm = ( { onSubmit } ) => {
 	const { stepData } = useSetupWizardStep( 'newsletter' );
 
-	const submitHandler = () => {
-		// Make sure it will run after the submit is done.
-		setTimeout( () => {
-			onSubmit();
-		} );
+	const [ email, setEmail ] = useState( stepData.admin_email );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ error, setError ] = useState( false );
+
+	const submitHandler = ( e ) => {
+		e.preventDefault();
+
+		const formData = new window.FormData( e.target );
+
+		const url = addQueryArgs(
+			e.target.action,
+			Object.fromEntries( formData )
+		);
+
+		// JSONP callback.
+		window[ SIGNUP_CALLBACK ] = ( json ) => {
+			setIsSubmitting( false );
+
+			if ( 'error' === json.result ) {
+				setError( json.msg );
+			} else {
+				onSubmit();
+			}
+		};
+
+		// Create and append JSONP script.
+		const script = document.createElement( 'script' );
+		script.src = `${ url }&c=${ SIGNUP_CALLBACK }`;
+
+		setIsSubmitting( true );
+		setError( false );
+		document.body.append( script );
 	};
 
 	return (
-		<form
-			action={ stepData.mc_url }
-			method="post"
-			target="_blank"
-			className="sensei-signup-form"
-			onSubmit={ submitHandler }
-		>
-			<input
-				type="hidden"
-				name={ `gdpr[${ stepData.gdpr_field }]` }
-				value="Y"
-			/>
+		<>
+			{ error && (
+				<Notice
+					status="error"
+					className="sensei-setup-wizard__submit-error"
+					isDismissible={ false }
+				>
+					{ error }
+				</Notice>
+			) }
+			<form
+				action={ stepData.mc_url }
+				method="GET"
+				className="sensei-signup-form"
+				onSubmit={ submitHandler }
+			>
+				<input
+					type="hidden"
+					name={ `gdpr[${ stepData.gdpr_field }]` }
+					value="Y"
+				/>
```

Comment:
Won't this default be possibly dangerous?

Reply:
Honestly, I didn't dig into this because it's how we use it currently. So I imagine it's not a big deal? 🤷‍♂️